R/na_if.R

```diff
@@ -0,0 +1,32 @@
+#' Convert values to NA
+#'
+#' @param x A vector to modify
+#' @param y What to replace with `NA`. Can be:
+#'
+#' - A vector whose type matches `x`, in that case [vctrs::vec_in()] is used
+#'   to determine which values are replace with `NA`.
+#' - A predicate function applied to `x` returning a logical vector, integer or character vector. See [vctrs::vec_slice<-()] for details.
+#'
+#' @examples
+#' x <- 1:10
+#'
+#' # turn specific values to NA
+#' na_if(x, c(6, 8))
+#'
+#' # or use a predicate
+#' na_if(x, ~ .x > 5)
+#'
+#' df <- data.frame(x = 1:10, y = 1:10)
+#' na_if(df, ~ .x$y > 8)
+#'
+#' @export
+na_if <- function(x, y) {
+  if (vec_is(y, ptype = y)) {
```

Comment:
I think this test should be the other way around — i.e. use `vec_in()` if `y` is not a function.

Reply:
That makes sense, but then we have to test for formulas first.